Makefile

```diff
@@ -102,3 +102,12 @@ container-push:
 .PHONY: docs
 docs:
 	$(MAKE) -C $(REPO_ROOT)/documentation build
+
+.PHONY: test-e2e
+test-e2e:
+	@echo "Beginning end-to-end tests";
+	@echo "Checking if the API is running...";
+	@curl --silent --output /dev/null localhost:8000 || (echo "Failed to connect to the API. Is it running?"; exit 1;)
```

Comment:
Can we start the api here? Similar to how we're doing it for `make test`

Reply:
I solved this by adding a new Make target called `run-test` which launches the API using the test config. Then, when the `integration-test` target is made, it has `run-test` as a dependency. But there is an issue I have been unable to solve involving a race condition, sometimes the API does not fully start before the `integration-test` target tries to run its tests, so it fails. 

I think these should be separate issues since the integration tests are now working, you just have to launch the API yourself on the side.